src/native/libs/System.Native/pal_io.c

```diff
@@ -766,6 +767,38 @@ int32_t SystemNative_SymLink(const char* target, const char* linkPath)
     return result;
 }
 
+uint64_t SystemNative_MakeDev(uint32_t major, uint32_t minor)
+{
+    return (uint64_t)makedev(major, minor);
+}
+
+void SystemNative_GetDeviceIdentifiers(uint64_t dev, uint32_t* major, uint32_t* minor)
+{
+    dev_t castedDev = (dev_t)dev;
+    *major = major(castedDev);
+    *minor = minor(castedDev);
+}
+
+int32_t SystemNative_MkNod(const char* pathName, int32_t mode, uint32_t major, uint32_t minor)
+{
+    dev_t dev = makedev(major, minor);
```

Comment:
This is failing in WASM:
```
##[error]src/native/libs/System.Native/pal_io.c(784,17): error GB8F1AD81: (NETCORE_ENGINEERING_TELEMETRY=Build) implicit conversion loses integer precision: 'unsigned long long' to 'dev_t' (aka 'unsigned int') [-Werror,-Wshorten-64-to-32]
      dev_t dev = makedev(major, minor);
            ~~~   ^~~~~~~~~~~~~~~~~~~~~
  /usr/local/emscripten/emsdk/upstream/emscripten/cache/sysroot/include/sys/sysmacros.h:12:37: note: expanded from macro 'makedev'
          (((y)&0xffffff00ULL) << 12) | \
          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~
  1 error generated.
```

I'll see if I can solve it with:
```
#if defined(TARGET_WASM)
    unsigned long long
#else
    dev_t
#endif
    dev = makedev(major, minor);
```

Thoughts, @steveisok?

Reply:
Didn't need this in the end. A simple explicit casting fixed the problem for all the platforms where the implicit conversion was causing problems.